### PR TITLE
Add reproducible builds

### PR DESCRIPTION
**Pull request by phklive**
This improves the reliability of the builds for the miden-node enabling more consistency and reproducibility.

closes: #251

### CODE REVIEW

**Comment by phklive** [comment]
@hackaugusto for now we are not using the matrix capabilities in the CI. Should we still keep the matrix structure for future proofing or should we switch to a simpler CI structure?

**Comment by hackaugusto** [comment]
> @hackaugusto for now we are not using the matrix capabilities in the CI. Should we still keep the matrix structure for future proofing or should we switch to a simpler CI structure?

I don't mind either way.

**Comment by phklive** [comment]
@hackaugusto should we also add doc and test in cargo make file? 

Considering that these can be ran without flags for now and easier: `cargo test` against `cargo make test`

What do you think?

**Comment by phklive** [comment]
> > @hackaugusto for now we are not using the matrix capabilities in the CI. Should we still keep the matrix structure for future proofing or should we switch to a simpler CI structure?
> 
> I don't mind either way.

Removed the matrix to simplify CI for now

**Review by hackaugusto — APPROVED** [review]
LGTM, thanks!

File: .gitignore [hunk 1]
```diff
@@ -8,7 +8,7 @@ target/
 
 # Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
 # More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
-Cargo.lock
+# Cargo.lock
```
Comment (bobbinth) [inline .gitignore:11]: Is this comment still relevant?

File: rust-toolchain [hunk 2]
```diff
@@ -0,0 +1 @@
+1.75
```
Comment (bobbinth) [inline rust-toolchain:1]: We should probably set MSRV in the workspace `Cargo.toml` file and I'm curious if there is a way to ensure that one and this one are in sync.

File: Makefile.toml
Comment (bobbinth) [inline Makefile.toml:1]: Should we add tasks for testing to this file as well?

Comment (phklive) [reply Makefile.toml:1]: We do have one here: https://github.com/0xPolygonMiden/miden-node/pull/267/files/43b7147397c498bfd44030bc75c819fa7bf80b47#diff-9375fd04332c86472d7be397ef09428cb86babd8826880a5835bd1d1c1bdbc08R30

Are you thinking of something else?

File: rust-toolchain (hunk 2 above)
Comment (phklive) [reply rust-toolchain:1]: Will add it to the cargo.toml, i can check but  I do not think that there is a standard way to check that both versions match.

File: .gitignore (hunk 1 above)
Comment (phklive) [reply .gitignore:11]: Will remove it.

File: Makefile.toml
Comment (bobbinth) [reply Makefile.toml:1]: Ah - I missed it! n/m

File: rust-toolchain (hunk 2 above)
Comment (phklive) [reply rust-toolchain:1]: After doing a quick research i have not found a standard way of checking that all crates match the version of the rust-toolchain file. Hence decided to create a PR to address that and fix your comments:
https://github.com/0xPolygonMiden/miden-node/pull/272

**Pull request merged**